firebase_admin/auth.py

```diff
@@ -94,10 +96,653 @@ def verify_id_token(id_token, app=None):
+                raise ValueError('Malformed photo URL: "{0}".'.format(photo_url))
+        except Exception:
+            raise ValueError('Malformed photo URL: "{0}".'.format(photo_url))
+
+    @classmethod
+    def validate_disabled(cls, disabled):
+        if not isinstance(disabled, bool):
+            raise ValueError(
+                'Invalid disabled status: "{0}". Disabled status must be '
+                'boolean.'.format(disabled))
+
+    @classmethod
+    def validate_delete_list(cls, delete_attr):
+        if not isinstance(delete_attr, list) or not delete_attr:
+            raise ValueError(
+                'Invalid delete list: "{0}". Delete list must be a '
+                'non-empty list.'.format(delete_attr))
+
+
+class _UserManager(object):
+    """Provides methods for interacting with the Google Identity Toolkit."""
+
+    _ID_TOOLKIT_URL = 'https://www.googleapis.com/identitytoolkit/v3/relyingparty/'
+
+    _VALIDATORS = {
+        'deleteAttribute' : _Validator.validate_delete_list,
+        'deleteProvider' : _Validator.validate_delete_list,
+        'disabled' : _Validator.validate_disabled,
+        'disableUser' : _Validator.validate_disabled,
+        'displayName' : _Validator.validate_display_name,
+        'email' : _Validator.validate_email,
+        'emailVerified' : _Validator.validate_email_verified,
+        'localId' : _Validator.validate_uid,
+        'password' : _Validator.validate_password,
+        'phoneNumber' : _Validator.validate_phone,
+        'photoUrl' : _Validator.validate_photo_url,
+    }
+
+    _CREATE_USER_FIELDS = {
+        'uid' : 'localId',
+        'display_name' : 'displayName',
+        'email' : 'email',
+        'email_verified' : 'emailVerified',
+        'phone_number' : 'phoneNumber',
+        'photo_url' : 'photoUrl',
+        'password' : 'password',
+        'disabled' : 'disabled',
+    }
+
+    _UPDATE_USER_FIELDS = {
+        'display_name' : 'displayName',
+        'email' : 'email',
+        'email_verified' : 'emailVerified',
+        'phone_number' : 'phoneNumber',
+        'photo_url' : 'photoUrl',
+        'password' : 'password',
+        'disabled' : 'disabled',
+    }
+
+    _REMOVABLE_FIELDS = {
+        'displayName' : 'DISPLAY_NAME',
+        'photoUrl' : 'PHOTO_URL'
+    }
+
+    _INTERNAL_ERROR = 'INTERNAL_ERROR'
+    _USER_NOT_FOUND_ERROR = 'USER_NOT_FOUND_ERROR'
+    _USER_CREATE_ERROR = 'USER_CREATE_ERROR'
+    _USER_UPDATE_ERROR = 'USER_UPDATE_ERROR'
+    _USER_DELETE_ERROR = 'USER_DELETE_ERROR'
+
+    def __init__(self, app):
+        g_credential = app.credential.get_credential()
+        session = transport.requests.AuthorizedSession(g_credential)
+        version_header = 'Python/Admin/{0}'.format(firebase_admin.__version__)
+        session.headers.update({'X-Client-Version': version_header})
+        self._session = session
+
+    def get_user(self, uid):
+        """Gets the user data corresponding to the specified user ID."""
```

Comment:
all these get users are basically the same except for the error message.  Does it make sense to have a shared implementation that takes in a parameter for the error messages?